### PR TITLE
LTD-4843 Add entity_type field on DenialsEntity model

### PR DESCRIPTION
**Pull request by markj0hnst0n**
### Aim

Create a choices field in DenialEntity model for entity_type with the correct options as per get_denial_entity_type method

[LTD-4843](https://uktrade.atlassian.net/browse/LTD-4843)

[LTD-4843]: https://uktrade.atlassian.net/browse/LTD-4843?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/external_data/models.py [hunk 1]
```diff
@@ -59,6 +59,14 @@ class DenialEntity(TimestampableModel):
         help_text="Reason why the denial was refused", blank=True, default="", null=True
     )
     spire_entity_id = models.IntegerField(help_text="Entity_id from spire for matching data", null=True)
+    entity_type_choices = (
+        ("end_user", "End-user"),
+        ("consignee", "Consignee"),
+        ("third_party", "Third-party"),
+    )
+    entity_type = models.TextField(
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply api/external_data/models.py:67]: This is what choices normally look like 
https://github.com/uktrade/lite-api/blob/910d0f824ae491392f9e72c4c06b61a42a9e9f7b/api/applications/enums.py#L202

need to separate them out from the model have a search around the code and you'll see some examples.

**Review by depsiatwal — COMMENTED** [review]
You will also need a migration file

./manage.py makemigrations

**Comment by markj0hnst0n** [comment]
made relevant changes made based on the comments

File: api/external_data/enums.py [hunk 2]
```diff
@@ -6,3 +9,9 @@ class DenialMatchCategory:
         (PARTIAL, "Partial"),
         (EXACT, "Exact"),
     ]
+
+
+class DenialEntityType(models.TextChoices):
+    END_USER = "END_USER", "End-user"
```
Comment (depsiatwal) [inline api/external_data/enums.py:15]: to be consistent with some of the other choices I would use low case _ 
so end_user

File: api/external_data/enums.py [hunk 3]
```diff
@@ -6,3 +9,9 @@ class DenialMatchCategory:
         (PARTIAL, "Partial"),
         (EXACT, "Exact"),
     ]
+
+
+class DenialEntityType(models.TextChoices):
+    end_user = "end_user", "End-user"
```
Comment (depsiatwal) [inline api/external_data/enums.py:15]: This should be like this 

END_USER = "end_user"
CONSIGNEE = "consignee"
THIRD_PARTY= "third_party"

Comment (markj0hnst0n) [reply api/external_data/enums.py]: I've implemented this hopefully.  I've used lower case i.e. end_user="End-user" as I think that's what was suggested previously

Comment (hnryjmes) [reply api/external_data/enums.py]: I'm now more confused than I was before I reviewed this PR

Comment (hnryjmes) [reply api/external_data/enums.py]: @depsiatwal can you give a clear example of style you would prefer so Mark can implement it how you like

Comment (hnryjmes) [reply api/external_data/enums.py]: I was just reading the docs, and it seems like there are different styles allowed depending on what the choices are being used for
https://docs.djangoproject.com/en/4.2/ref/models/fields/#choices

Comment (hnryjmes) [reply api/external_data/enums.py]: @depsiatwal you can ignore my previous comment, I was confused about the different styles, you were perfectly clear in what you said

Comment (depsiatwal) [reply api/external_data/enums.py]: actually lets make it  like this so it's consistent  
https://github.com/uktrade/lite-api/blob/910d0f824ae491392f9e72c4c06b61a42a9e9f7b/api/parties/enums.py#L1
That might be ho you originally had it lol

Comment (markj0hnst0n) [reply api/external_data/enums.py]: lol with so many commits you're making me look good here gurdeep!

File: api/external_data/enums.py [hunk 4]
```diff
@@ -6,3 +9,9 @@ class DenialMatchCategory:
         (PARTIAL, "Partial"),
         (EXACT, "Exact"),
     ]
+
+
+class DenialEntityType(models.TextChoices):
+    end_user = "End-user"
+    consignee = "Consignee"
+    third_party = "Third-party"
```
[the remark this answers is not in the document]
Comment (markj0hnst0n) [reply api/external_data/enums.py]: thanks arun!

**Comment by markj0hnst0n** [comment]
have made change to this again

**Review by saruniitr — APPROVED** [review]

**Pull request merged**